src/pages/Calculator.tsx

```diff
@@ -67,6 +68,19 @@ export const Calculator = (): React.ReactElement => {
     useQueryDataIfPresent(query, migratedPreviousData),
   )
 
+  const { t } = useTranslation()
+
+  const [editScenario, setEditScenario] = useState(
```

Comment:
Having the logic for what is visible be separated from the data makes it hard to reason about.

It looks to me that both editScenario and editInteractionType are (approximately) equal to "is a preset selected"

Could we add the selected preset to CalculatorData and use this directly? It would also cause share links to show the same scenario.

Reply:
editInteractionType is "does the user need to set oneTime, workplace, repeated or partner right now?" which is strongly tied to "is custom" but not equivalent.

editScenario is "no preset nor custom"

editScenario seems like it could be removed by moving scenarioName/setScenarioName to CalculatorData. I'll look into it.

Reply:
> Could we add the selected preset to CalculatorData and use this directly? It would also cause share links to show the same scenario.

Done. Also cleaned out the edit{InteractionType,Scenario} from Calculator.tsx.